crates/ln-dlc-node/src/node/ln_channel.rs

```diff
@@ -60,53 +57,90 @@ impl<P> Node<P> {
+        let channel = channels
             .iter()
-            .find(|channel| channel.channel_id == channel_id);
-
-        match channels_to_close {
-            Some(cd) => {
-                if force_close {
-                    tracing::debug!(
-                        "Force closing channel {} with peer {} ",
-                        hex::encode(cd.channel_id),
-                        cd.counterparty.node_id
-                    );
-                    channel_manager
-                        .force_close_broadcasting_latest_txn(
-                            &cd.channel_id,
-                            &cd.counterparty.node_id,
-                        )
-                        .map_err(|e| anyhow!("Could not force close channel {e:?}"))
-                } else {
-                    if let Some(SubChannel { state, .. }) = self.list_dlc_channels()?.first() {
-                        if *state != SubChannelState::OffChainClosed {
-                            // this is a safeguard to not close a ln channel with
-                            // an open dlc, as we would lose the funds otherwise.
-
-                            //TODO: Ideally we would start to collaborative close the dlc
-                            // channel here first and then close the ln
-                            // channel once done.
-                            bail!("Cannot close ln channel with an open dlc.");
-                        }
-                    }
-
-                    tracing::info!(
-                        "Collaboratively closing channel {} with peer {} ",
-                        hex::encode(cd.channel_id),
-                        cd.counterparty.node_id
-                    );
-                    channel_manager
-                        .close_channel(&cd.channel_id, &cd.counterparty.node_id)
-                        .map_err(|e| anyhow!("Could not collaboratively close channel {e:?}"))
-                }
-            }
-            None => {
-                bail!("No channel found with ID {}", hex::encode(channel_id))
-            }
+            .find(|channel| channel.channel_id == channel_id)
+            .with_context(|| format!("Cannot close non-existent channel {channel_id_str}"))?;
+
+        if force_close {
+            self.force_close_channel(channel)?;
+        } else {
+            self.collab_close_channel(channel)?;
         }
+
+        Ok(())
+    }
+
+    fn collab_close_channel(&self, channel: &ChannelDetails) -> Result<()> {
+        let channel_id = channel.channel_id;
+        let channel_id_str = hex::encode(channel_id);
+        let peer = channel.counterparty.node_id;
+
+        tracing::info!(channel_id = %hex::encode(channel_id), %peer, "Collaboratively closing channel");
+
+        self.is_safe_to_close_ln_channel_collaboratively(&channel_id)
+            .with_context(|| {
+                format!("Could not collaboratively close LN channel {channel_id_str}")
```

Comment:
🤔 Probably add that the existing DLC should be closed (settled) before the channel can be collaboratively closed? IMHO that would be more helpful as an error message.